### PR TITLE
[Maintanence + DX] - delete old queries and POC gql-tada

### LINKED ISSUE #10185: [DX] Generate auto types from graphql and rest api
Related:
- https://github.com/kodadot/nft-gallery/issues/8500
- https://github.com/kodadot/nft-gallery/issues/4544

Graphql queries are higher priorities.

### PR DESCRIPTION
**Pull request by daiagi**
**Thank you for your contribution** to the [Koda - Generative Art Marketplace](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [x] Feature
- [x] Refactoring / Maintenance / Remove dead code

## Needs QA check

- @kodadot/qa-guild please review

## Context

- [x] Closes #10185

#### Did your issue had any of the "$" label on it?

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8)

### CODE REVIEW

**Comment by daiagi** [comment]
this has grown larger then i intended


1. I have deleted all unused graphql queries
2. i have deleted some dead pages and their associated components => delete more dead queries
3. I have converted some queries to types ones, but it's a large job, so let's continue in a follow up


@Jarsen136 @preschian

**Comment by prury** [comment]
Playwright detection: Collection banner/names are not loading

image

**Comment by daiagi** [comment]
@prury 


yes i have noticed

fixed

**Comment by preschian** [comment]
> 3\. but it's a large job, so let's continue in a follow up

Nice 👍🏻
Yes, there is no need to migrate it to one PR

Can you summarise a little bit of how to use it and the tradeoff? And mention internal dev

**Comment by daiagi** [comment]
sure

first let's break this PR apart so it doesn't look so scary and big

1. 56 deleted graphql files (old and unused queries, spring cleanup)
2. 16 deleted `.vue` and `.ts` files. these are dead pages and their associated sub components (hot, sales, spotlight pages)
3. 1 deleted unused composable ([composables/drop/useCheckReferenDumVote.ts](https://github.com/kodadot/nft-gallery/pull/10197/files?file-filters%5B%5D=.ts&show-deleted-files=true&show-viewed-files=true#diff-7de5983aaef3a5f15810c23e65db6e80f7c716d7c3db891930854e92f7b571d4))
4.  3 deleted types i've created in #10176 

so far 76 deleted files

5. 8 new `.ts` files  typed fragments
6. 4 new `d.ts` files that contain the typings of the graphql clients, 1 file for the functions that take in a query text and return typed `DocumentNode`
7. 6 new typed queried

19 new files

so far 95 files

8. 6 edited componenets
9. 4 misclenious (package.json, tsconfig and such)

105 files

**Comment by vikiival** [comment]
If it is suitable -
Aka for queries that already use subsquid/general it is ok to use typed queries

However for specific chains I would resolve Query path - or better checl when and why it was added back then

**Comment by daiagi** [comment]
>if it is suitable -
Aka for queries that already use subsquid/general it is ok to use typed queries

However for specific chains I would resolve Query path - or better checl when and why it was added back then


i agree, this is a good tradeoff between both approaches


therefore, i will delete the typed clients that are chain specific, and will call `https://ahp.gql.api.kodadot.xyz/` GENERAL

**Comment by daiagi** [comment]
cc @preschian @Jarsen136

File: components/collection/CollectionHeader/CollectionBanner.vue [hunk 1]
```diff
@@ -72,11 +73,17 @@ watchEffect(async () => {
   const image = collection?.meta?.image
   const name = collection?.name
 
+  console.log('debug collection.metdadta', metadata)
+  console.log('debug collection.meta.image', image)
+  console.log('debug collection.name', name)
```
Comment (Jarsen136) [inline components/collection/CollectionHeader/CollectionBanner.vue:78]: Could you remove these `console.log` ?

File: .gitignore [hunk 2]
```diff
@@ -95,3 +95,7 @@ sitemap.xml
 /test-results/
 /playwright-report/
 /playwright/.cache/
+
+
+queries/clients/types
```
Comment (Jarsen136) [inline .gitignore:100]: It does not work. I still see the types file on the commits.

image

**Review by Jarsen136 — APPROVED** [review]

**Review by Jarsen136 — COMMENTED** [review]
.

**Review by Jarsen136 — CHANGES_REQUESTED** [review]
as comment above

**Comment by daiagi** [comment]
thx @Jarsen136 

fixed

**Review by Jarsen136 — APPROVED** [review]

File: tsconfig.json [hunk 3]
```diff
@@ -22,6 +22,18 @@
...
       "vitest/globals"
+    ],
+    "plugins": [
+      {
+        "name": "@0no-co/graphqlsp",
+        "schemas": [
+          {
+            "name": "ahp",
+            "schema": "https://ahp.gql.api.kodadot.xyz/",
+            "tadaOutputLocation": "./queries/clients/types/ahp.d.ts"
```
Comment (preschian) [inline tsconfig.json:33]: The folder doesn't exist; it won't create the types on launch. Adjust it and ignore `*.d.ts` files

```suggestion
            "tadaOutputLocation": "./queries/clients/ahp.d.ts"
```

File: components/collection/CollectionHeader/CollectionBanner.vue [hunk 4]
```diff
@@ -93,21 +95,21 @@ watchEffect(async () => {
 useSeoMeta({
   title: collectionName,
   description: () =>
-    convertMarkdownToText(data.value?.collectionEntity.meta?.description),
+    convertMarkdownToText(data.value?.collectionEntity?.meta?.description),
```
Comment (preschian) [inline components/collection/CollectionHeader/CollectionBanner.vue:98]: I'm getting this on `convertMarkdownToText()`

```
Argument of type 'string | null | undefined' is not assignable to parameter of type 'string | undefined'.
```

**Review by preschian — APPROVED** [review]
sorry for the delay, otherwise lgtm

Additionals:
- remove `~/components/spotlight` on `nuxt.config.ts`
- nitpick: separate `gql.tada` queries from raw graphql queries

Screenshot 2024-05-13 at 11 33 26

**Comment by daiagi** [comment]
@preschian  Thank you

All done

File: queries/fragments/typed/nftSubsquid.ts
Comment (vikiival) [inline queries/fragments/typed/nftSubsquid.ts:1]: Prob delete as it is same as `nft.ts`

File: queries/fragments/typed/nft.ts
Comment (vikiival) [inline queries/fragments/typed/nft.ts:1]: add `updatedAt` to be able to remove [nftSubsquid.ts](https://github.com/kodadot/nft-gallery/pull/10197/files#diff-c6737b7326d724e5a4bd9d632c5cacc43cbc57e29660d83854ef84475f83de5e)

**Review by vikiival — COMMENTED** [review]
~

**Comment by vikiival** [comment]
Please @daiagi, @Jarsen136 and @prury go through app and make sure that every page and chain work as intended :)

**Comment by daiagi** [comment]
@vikiival 



>Prob delete as it is same as nft.ts
>add updatedAt to be able to remove


done

**Comment by daiagi** [comment]
i have scanned throuh

- home page (all chains)
- explore collections and items (all chains)
- my profile page
- drops home page
- single ahk drop page
- create pages
- single nfty page (all chains)
- single collection page (all chains)



**so far 0 graphql detected**

**Comment by daiagi** [comment]
>How about trasactions like create collection,
list/buy NFT ?

i did buy and list 1 ahp nft

**Comment by Jarsen136** [comment]
I went through the website and did not find anything abnormal.

**Comment by prury** [comment]
> > How about trasactions like create collection,
> > list/buy NFT ?
> 
> i did buy and list 1 ahp nft

i will perform transaction tests as well trough the day

**Comment by prury** [comment]
NFT Transfer :white_check_mark: 
Change Price :white_check_mark: 
Collection Creation :white_check_mark: 
NFT Creation(List, copies) :white_check_mark: 
Mass buy :white_check_mark: 
Mass list :white_check_mark: 
Drop Mint :white_check_mark:

**Comment by prury** [comment]
failing test can be ignored

**Comment by prury** [comment]
i've resolved conversations, merging now

**Pull request merged**